.github/workflows/full_kubeflow_integration_test.yaml

```
@@ -21,139 +21,161 @@ jobs:
+
+    - name: Wait for All Pods to be Ready
+      run: kubectl wait --for=condition=Ready pods --all --all-namespaces --timeout 300s --field-selector=status.phase!=Succeeded
+
+    - name: Install Katib - Prepare
       run: |
-        # Katib
         sudo apt-get install -y apparmor-profiles
         sudo apparmor_parser -R /etc/apparmor.d/usr.sbin.mysqld
-        cd apps/katib/upstream && kustomize build installs/katib-with-kubeflow | kubectl apply -f - && cd ../../../
-        kubectl wait --for=condition=Available deployment/katib-controller -n kubeflow --timeout=300s
-        kubectl wait --for=condition=Available deployment/katib-mysql -n kubeflow --timeout=300s
-        kubectl label namespace $KF_PROFILE katib.kubeflow.org/metrics-collector-injection=enabled --overwrite
 
-        # Other ML components
-        ./tests/gh-actions/install_training_operator.sh
-        ./tests/gh-actions/install_knative.sh
-        ./tests/gh-actions/install_kserve.sh
-        chmod u+x tests/gh-actions/spark_*.sh && ./tests/gh-actions/spark_install.sh
+    - name: Install Katib
+      run: cd apps/katib/upstream && kustomize build installs/katib-with-kubeflow | kubectl apply -f - && cd ../../../
+
+    - name: Wait for Katib Controller
+      run: kubectl wait --for=condition=Available deployment/katib-controller -n kubeflow --timeout=300s
+
+    - name: Wait for Katib MySQL
+      run: kubectl wait --for=condition=Available deployment/katib-mysql -n kubeflow --timeout=300s
+
+    - name: Configure Katib in Profile
+      run: kubectl label namespace $KF_PROFILE katib.kubeflow.org/metrics-collector-injection=enabled --overwrite
+
+    - name: Install Training Operator
+      run: ./tests/gh-actions/install_training_operator.sh
+
+    - name: Install Knative
+      run: ./tests/gh-actions/install_knative.sh
+
+    - name: Install KServe
+      run: ./tests/gh-actions/install_kserve.sh
+
+    - name: Install Spark
+      run: chmod u+x tests/gh-actions/spark_*.sh && ./tests/gh-actions/spark_install.sh
 
-        # Dependencies
-        pip install pytest kubernetes kfp==2.11.0 kserve pytest-timeout pyyaml requests
+    - name: Install Dependencies
+      run: pip install pytest kubernetes kfp==2.11.0 kserve pytest-timeout pyyaml requests
 
     - name: Port-forward the istio-ingress gateway
-      run: |
-        GATEWAY=$(kubectl get svc -n istio-system -l app=istio-ingressgateway -o jsonpath='{.items[0].metadata.name}')
-        nohup kubectl port-forward -n istio-system svc/$GATEWAY 8080:80 &
-        while ! curl -s localhost:8080 > /dev/null; do sleep 1; done
+      run: ./tests/gh-actions/port_forward_gateway.sh
 
-    - name: Test Dex Authentication
-      run: |
-        chmod +x tests/gh-actions/test_dex_auth.sh && ./tests/gh-actions/test_dex_auth.sh
+    - name: Setup OAuth2 and Dex Credentials
+      run: chmod +x tests/gh-actions/oauth2_dex_credentials.sh && ./tests/gh-actions/oauth2_dex_credentials.sh
```

lets keep this PR simple, just use the default password etc for now.
```suggestion
    #- name: Setup OAuth2 and Dex Credentials
    #  run: chmod +x tests/gh-actions/oauth2_dex_credentials.sh && ./tests/gh-actions/oauth2_dex_credentials.sh
```